app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -171,9 +171,9 @@ if [ -f /etc/redhat-release ]; then
     # Configure subscription-manager
     test -f $RHSM_CFG.bak || cp $RHSM_CFG $RHSM_CFG.bak
     subscription-manager config \
-      --server.hostname="<%= @rhsm_url.host %>" \
-      --server.port="<%= @rhsm_url.port %>" \
-      --server.prefix="<%= @rhsm_url.path %>" \
+      --server.hostname="<%= @rhsm_url.host if @rhsm_url %>" \
+      --server.port="<%= @rhsm_url.port if @rhsm_url %>" \
+      --server.prefix="<%= @rhsm_url.path if @rhsm_url %>" \
```

Comment:
I'd like a safe navigation better here, it is more clear then that the params will be empty is that what we want, for should we omit them altogether `unless @rhsm_url`?
```suggestion
      --server.hostname="<%= @rhsm_url&.host %>" \
      --server.port="<%= @rhsm_url&.port %>" \
      --server.prefix="<%= @rhsm_url%.path %>" \
```

Reply:
You can't use safe operator in templates (in safe mode), you'll get error:
```
There was an error rendering the template: wrong number of arguments (given 2, expected 1)

2022-01-19T14:31:00 [I|app|cba3b039] Backtrace for 'Error rendering the  template' error (ArgumentError): wrong number of arguments (given 2, expected 1)
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/safemode-1.3.6/lib/safemode/parser.rb:35:in `process_call'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/ruby2ruby-2.4.4/lib/ruby2ruby.rb:308:in `process_safe_call'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/sexp_processor-4.16.0/lib/sexp_processor.rb:269:in `block (2 levels) in process'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/sexp_processor-4.16.0/lib/sexp_processor.rb:324:in `error_handler'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/sexp_processor-4.16.0/lib/sexp_processor.rb:268:in `block in process'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/sexp_processor-4.16.0/lib/sexp_processor.rb:388:in `in_context'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/sexp_processor-4.16.0/lib/sexp_processor.rb:245:in `process'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/ruby2ruby-2.4.4/lib/ruby2ruby.rb:207:in `block in process_block'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/ruby2ruby-2.4.4/lib/ruby2ruby.rb:206:in `map'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/ruby2ruby-2.4.4/lib/ruby2ruby.rb:206:in `process_block'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/sexp_processor-4.16.0/lib/sexp_processor.rb:269:in `block (2 levels) in process'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/sexp_processor-4.16.0/lib/sexp_processor.rb:324:in `error_handler'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/sexp_processor-4.16.0/lib/sexp_processor.rb:268:in `block in process'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/sexp_processor-4.16.0/lib/sexp_processor.rb:388:in `in_context'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/sexp_processor-4.16.0/lib/sexp_processor.rb:245:in `process'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/safemode-1.3.6/lib/safemode/parser.rb:10:in `jail'
 cba3b039 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/safemode-1.3.6/lib/safemode.rb:49:in `eval'
 cba3b039 | /home/vagrant/foreman/app/services/foreman/renderer/safe_mode_renderer.rb:7:in `render'
 cba3b039 | /home/vagrant/foreman/app/services/foreman/renderer/base_renderer.rb:18:in `render'
 cba3b039 | /home/vagrant/foreman/app/models/template.rb:172:in `render'
 cba3b039 | /home/vagrant/foreman/app/controllers/templates_controller.rb:119:in `safe_render'
 cba3b039 | /home/vagrant/foreman/app/controllers/templates_controller.rb:99:in `preview'
```

It's maybe worth looking into it but I'm not sure it's something I want to do as part of this PR

Reply:
Ugh, didn't know ok then :+1: